src/main/java/seedu/address/logic/commands/EditClientCommand.java

```diff
@@ -109,13 +108,18 @@ public CommandResult execute(Model model) throws CommandException {
         if (personToEdit.getCar() != null) {
             Car carToEdit = personToEdit.getCar();
             editedCar = createEditedCar(carToEdit, editCarDescriptor);
-            if (carToEdit.equals(editedCar) && model.hasCar(editedCar) && isCarEdited) {
+            if ((carToEdit.equals(editedCar) || model.hasCar(editedCar)) && isCarEdited) {
                 throw new CommandException(MESSAGE_DUPLICATE_CAR);
             }
         }
 
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor, editedCar);
-        if (personToEdit.equals(editedPerson) && model.hasPerson(editedPerson) && isPersonEdited) {
+
+        if (personToEdit.equals(editedPerson) && isPersonEdited) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+        }
+
+        if ((personToEdit.getName() != editedPerson.getName()) && model.hasPerson(editedPerson) && isPersonEdited) {
```

Comment:
(personToEdit.getName() != editedPerson.getName()) is redundant since model.hasPerson checks for it